assets/js/modules/analytics-4/utils/report-validation.test.js

```diff
@@ -157,4 +158,135 @@ describe( 'Analytics 4 Reporting API validation', () => {
 			).toBe( false );
 		} );
 	} );
+
+	describe( 'isValidOrders', () => {
+		it( 'should return TRUE if an array of valid order objects is passed', () => {
+			expect(
+				isValidOrders( [
+					{
+						metric: {
+							metricName: 'totalUsers',
+						},
+						desc: false,
+					},
+					{
+						metric: {
+							metricName: 'sessions',
+						},
+						desc: true,
+					},
+					{
+						dimension: {
+							dimensionName: 'date',
+						},
+						desc: true,
+					},
+					{
+						dimension: {
+							dimensionName: 'sessionDefaultChannelGrouping',
+						},
+					},
+				] )
+			).toBeTruthy();
```

Comment:
We need to use `.toBe( true )` because `.toBeTruthy()` doesn't fail the test if non-empty value is passed. So, I think  it would be better to be more precise.

Reply:
Good point. I had used `toBeTruthy()` as that's what the existing UA `isValidOrders()` tests use and I wanted to be consistent. I didn't realise that in fact the UA `isValidOrders()` doesn't always return a boolean and can in fact return  the result of `String.prototype.match()`:

https://github.com/google/site-kit-wp/blob/5ef7abacd409518ad9dfe49f553e9a841e20383e/assets/js/util/report-validation.js#L82-L85

I've created an issue to update UA `isValidOrders()` to always return a boolean, although it's pretty low priority. https://github.com/google/site-kit-wp/issues/6553